### ISSUE TITLE
`DD_CRI_SOCKET_PATH` not set in GKE Autopilot 

### ISSUE THREAD

**Issue by valkum** [DataDog/helm-charts#1188]
**Describe what happened:**
When the agent is new enough (supports autodiscovery), the helm chart does not set `DD_CRI_SOCKET_PATH` in GKE Autopilot.

**Describe what you expected:**
`DD_CRI_SOCKET_PATH` to be set for the agent containers.

**Cause:**
`_containers-common-env.yaml` will [check](https://github.com/DataDog/helm-charts/blob/main/charts/datadog/templates/_containers-common-env.yaml#L86) `agent-has-env-ad`, if it is true, it will only set `DD_CRI_SOCKET_PATH` if `.Values.datadog.criSocketPath` is [set](https://github.com/DataDog/helm-charts/blob/main/charts/datadog/templates/_containers-common-env.yaml#L107C7-L107C7).
If `agent-has-env-ad` is false, it additionally [checks](https://github.com/DataDog/helm-charts/blob/main/charts/datadog/templates/_containers-common-env.yaml#L107C7-L107C7) if this is running in Autopilot `.Values.providers.gke.autopilot`.
The containerd docs say `DD_CRI_SOCKET_PATH` is needed to enable the integration.

**Additional environment details (Operating System, Cloud provider, etc):**
GKE Autopilot

**Comment by rpriyanshu9** [comment]
Hi @valkum, pls correct me if I'm wrong, but are you saying that the `DD_CRI_SOCKET_PATH` is not set even when you have `.Values.providers.gke.autopilot` enabled?

Maybe you can explicitly set `.Values.datadog.criSocketPath` to the standard value defined [here](https://github.com/DataDog/helm-charts/blob/main/charts/datadog/README.md#cri-integration) and make it work?

**Comment by valkum** [comment]
Yes, when `.Values.providers.gke.autopilot` is `true`, and the agent version is latest or larger than `v7.27`, `DD_CRI_SOCKET_PATH` is not set.

We currently set `.Values.datadog.criSocketPath` to work around this issue. We expected this to be set automatically even if we use an agent version of at least `v7.27`.

**Comment by rpriyanshu9** [comment]
@valkum Yup, it won't be set. If you check [here](https://github.com/DataDog/helm-charts/blob/da2ef01812419431652b0432edb615aa94131474/charts/datadog/templates/_helpers.tpl#L46-L67) - on using datadog-agent's latest or any version >= `7.27`, this block of code will always evaluate to `true` and because of that the logic never enters this `else` clause [here](https://github.com/DataDog/helm-charts/blob/da2ef01812419431652b0432edb615aa94131474/charts/datadog/templates/_containers-common-env.yaml#L105). 

I might be wrong, I encourage you to verify what [agent-has-env-ad](https://github.com/DataDog/helm-charts/blob/da2ef01812419431652b0432edb615aa94131474/charts/datadog/templates/_helpers.tpl#L46-L67) results in if you are using the latest version of dd-agent.

This will help - https://helm-playground.com/

**Comment by rpriyanshu9** [comment]
I was thinking of raising a PR for configuring docker and containerd socket's default values in `values.yaml`, but since they're both mutually exclusive, the maintainers won't approve it. The idea behind leaving it to the end user would be to give more control to them on configuring these values.

**Comment by valkum** [comment]
Yeah. The question is: 
Is the documentation on for the `containerd` integration wrong and from 7.27 onwards setting `DD_CRI_SOCKET_PATH` should no longer be needed or should the true arm also get the same treatment as the false arm with:
```yaml
{{- if or .Values.providers.gke.autopilot .Values.datadog.criSocketPath }}
- name: DD_CRI_SOCKET_PATH
  value: {{ print "/host/" (include "datadog.dockerOrCriSocketPath" .) | clean }}
{{- else }}
```
